articles/how-to-choose-levels-of-logging/index.md

```diff
@@ -0,0 +1,65 @@
+Application logging is one of the most essential practices a developer can implement in their code to facilitate production support. Entries in log files contain essential information, including a timestamp, contextual information, and messages. But, log messages are not quite easy to understand.
+
+Most log management solutions that collect data and accept data from various sources have different log structures. Therefore, it may be challenging to understand them. It is crucial to keep meaningful log levels as we develop applications. This will ensure that anyone who reads the logs can make meaning of the intended message. It is essential to use the [best logging practices](https://www.scalyr.com/blog/the-10-commandments-of-logging/). This article gives a historical background of logging levels and describes different logging levels.
+
+### What is a logging level?
+
+A logging level is a way of classifying the entries in your log file in terms of urgency. Classifying helps filter your log files during search and control the amount of information in your logs. Sometimes, categorizing may require you to balance storage use. You may want to capture every detail useful in troubleshooting. Yet, it may use up storage causing your production code to slow down. While implementing logging levels, you do not necessarily need to know everything that happens in production. You can outline how to identify warning conditions or problems for troubleshooting purposes.
+
+### Background of logging levels
+
+Logging levels distinguish various log events from each other. They are a way of filtering important information about your system's state and reduce information noise or alert fatigue.
+
+Log levels were introduced in the 1980s in a project named [Sendmail](https://en.wikipedia.org/wiki/Sendmail). The project owners needed a logging solution that eventually led to the creation of a Syslog server and the idea of severity levels.
+
+Eric Allman developed the Sendmail project that eventually needed a logging solution. This is where the System Logging Protocol ([Syslog](https://www.paessler.com/it-explained/syslog#)) was introduced. Ultimately, Syslog was embraced by numerous applications and has become a standard protocol for sending system log and event messages to the Syslog server.
```

Comment:
You have some repetition here. The line: `Eric Allman developed the Sendmail project that eventually needed a logging solution.` repeats line 13.

Also, provide the Syslog link and full form of the word in the first appearance of the word.